test/dartdoc_test.dart

```diff
@@ -406,5 +406,19 @@ void main() {
         expect(e is DartdocFailure, isTrue);
       }
     });
+
+    test('rel canonical prefix does not include base href', () async {
+      final String prefix = 'foo.bar/baz';
+      Dartdoc dartdoc = await buildDartdoc(
+          ['--rel-canonical-prefix', prefix], testPackageMinimumDir, tempDir);
+      await dartdoc.generateDocsBase();
+
+      File file = File(path.join(tempDir.path, 'small', 'small-library.html'));
+      expect(file.existsSync(), isTrue);
+      expect(
+          file.readAsStringSync(),
+          contains(
+              '<link rel="canonical" href="$prefix/small/small-library.html">'));
```

Comment:
This seems like it should work but I would suggest verifying it for links at root and two levels down in the directory structure.

Reply:
Added another file to be checked in the test

Reply:
Oops, found a mistake in the test, will push up another commit.